src/BloomExe/Book/Book.cs

```diff
@@ -2093,11 +2129,15 @@ public void PrepareForEditing()
 			UpdateEditableAreasOfElement(OurHtmlDom);
 		}
 
+		public void UpdateEditableAreasOfElement(HtmlDom dom)
+		{
+			UpdateEditableAreasOfElement(dom.RawDom.DocumentElement);
+		}
 		/// <summary>
 		/// This is called both for the whole book, and for individual pages when the user uses Origami to make changes to the layout of the page
 		/// </summary>
 		/// <param name="elementToUpdate"></param>
-		public void UpdateEditableAreasOfElement(HtmlDom elementToUpdate)
+		public void UpdateEditableAreasOfElement(XmlElement elementToUpdate)
```

Comment:
Perhaps if you are able to get this xml manipulation encapsulated in html dom, you can remove this retrograde change :-)